src/segment.ml

```diff
@@ -180,7 +180,8 @@ type t = {
   push : bool ;
   window : int ;
   options : tcp_option list ;
-  payload : Cstruct.t ;
+  payload : Cstruct.t list ;
+  payload_len : int ;
```

Comment:
The rational behind this is: a `rope` can compiles its structure to a list of `Cstruct.t`. It can also **allocate** a new `Cstruct.t` and `blit` everything inside. To avoid any allocations in `utcp`, it's preferable to compiles a `rope` to a list of `Cstruct.t`. This diff updates the interface of `utcp` where `recv` returns a list of `Cstruct.t` instead of a simple `Cstruct.t` but we are sure that all them are **not allocated** by `utcp`.